bodhi/tests/server/tasks/test_updates.py

```diff
@@ -304,6 +306,91 @@ def test_update_not_found(self, work_on_bugs, fetch_test_cases, sleep):
+            with mock.patch('bodhi.server.models.util.greenwave_api_post') as mock_greenwave:
+                greenwave_response = {
+                    'policies_satisfied': False,
+                    'summary': 'what have you done‽',
+                    'applicable_policies': ['taskotron_release_critical_tasks'],
+                    'unsatisfied_requirements': [
+                        {'testcase': 'dist.rpmdeplint',
+                         'item': {'item': 'bodhi-2.0-1.fc17', 'type': 'koji_build'},
+                         'type': 'test-result-missing', 'scenario': None},
+                        {'testcase': 'dist.rpmdeplint',
+                         'item': {'item': update.alias, 'type': 'bodhi_update'},
+                         'type': 'test-result-missing', 'scenario': None}]}
+                mock_greenwave.return_value = greenwave_response
+                h.run(api_version=1,
+                      data={
+                          'action': 'edit',
+                          'update': {'alias': update.alias, 'builds': [{'nvr': 'bodhi-2.0-1.fc17'}],
+                                     'user': {'name': 'bodhi'}, 'status': str(update.status),
+                                     'request': str(update.request)}, 'new_bugs': []})
+
+        self.assertEqual(update.status, models.UpdateStatus.testing)
+        self.assertEqual(update.test_gating_status, models.TestGatingStatus.failed)
+
+    @mock.patch('bodhi.server.tasks.updates.UpdatesHandler.fetch_test_cases')
+    @mock.patch('bodhi.server.tasks.updates.UpdatesHandler.work_on_bugs')
+    def test_rawhide_update_edit_stays_pending(self, work_on_bugs, fetch_test_cases, sleep):
+        """
+        Assert that a pending rawhide update that was edited does not get moved to testing if the
+        updates builds are not signed.
+        """
+        update = models.Build.query.filter_by(nvr='bodhi-2.0-1.fc17').one().update
+        update.status = models.UpdateStatus.pending
+        update.release.composed_by_bodhi = False
+        update.builds[0].signed = False
+
+        h = updates.UpdatesHandler()
+        h.db_factory = base.TransactionalSessionMaker(self.Session)
+        h.run(api_version=1,
+              data={
+                  'action': 'edit',
+                  'update': {'alias': update.alias, 'builds': [{'nvr': 'bodhi-2.0-1.fc17'}],
+                             'user': {'name': 'bodhi'}, 'status': str(update.status),
+                             'request': str(update.request)}, 'new_bugs': []})
+
+        self.assertEqual(update.status, models.UpdateStatus.pending)
+
+    @mock.patch('bodhi.server.tasks.updates.UpdatesHandler.fetch_test_cases')
+    @mock.patch('bodhi.server.tasks.updates.UpdatesHandler.work_on_bugs')
+    def test_not_rawhide_update_signed_stays_pending(self, work_on_bugs, fetch_test_cases, sleep):
+        """
+        Assert that a pending rawhide update that was edited does not get moved to testing if the
```

Comment:
This seems to contradict the method name and code, i.e. it isn't about Rawhide updates, and the build is signed.